graphql/schema.graphql

```diff
@@ -90,6 +90,80 @@ type Mutation {
   Edit schedule and Plug and Play Compnay 3's and 4's Algorithms
   """
   editCourse(input: EditCourseInput): Response!
+    @deprecated(reason: "Use updateSchedule instead")
+  """
+  Update an entire schdule.
+  """
+  updateSchedule(input: UpdateScheduleInput!): UpdateScheduleResponse!
+}
+
+input UpdateScheduleInput {
+  """
+  ID of the schedule to update. If not given, the current schedule will be updated.
+  """
+  id: ID
+  """
+  The updated courses
+  """
+  courses: [CourseSectionInput!]!
+  """
+  Whether to perform validation on the backend through algorithm 1.
+  """
+  skipValidation: Boolean!
+  """
+  Which algorithm to use. 1 is the default. If COMPANY2 is selected then validation will not be performed regardless of skipValidation.
+  """
+  validation: Company!
```

Comment:
If C4 Alg 1 doesn't validate anyways why do we need to specify this attribute? Can't we just use `skipValidation` for this?

Reply:
My thinking was so frontend can more carefully control the behaviour of the backend response and UI respectively.

Reply:
Okay, doesnt affect how it works so I have no concerns